qt6/gen_qobject.cpp

```diff
@@ -36,7 +36,13 @@ void miqt_exec_callback_QObject_disconnectNotify(QObject*, intptr_t, QMetaMethod
 } /* extern C */
 #endif
 
+// This method's return type was changed from non-const to const in Qt 6.9
+#if QT_VERSION >= QT_VERSION_CHECK(6,9,0)
```

Comment:
can technically return the const pointer for all qt versions

Reply:
Isn't this a type mismatch if the underlying type isn't `const`? And couldn't it lead to undefined behavior on Qt versions under 6.9?

Ultimately, this is a stop-gap of sorts. Once we start building on a version at or above Qt 6.9, the patches in `config-allowlist.go` can go away. (It probably makes sense to leave the remaining code... just in case. 😅)

Reply:
> Once we start building

An interesting approach would actually be to generate the bindings for each qt version separately and use the `#if` trick for any "new" features - ie one would start with 6.0 generating all of that, then gradually add each version as it happens, hidden behind `#if` - then the mappings could / would work with any qt version that the user happens to have installed.

> Isn't this a type mismatch

you can assign a non-const pointer to a const pointer of the same underlying type without issue, [in general](https://en.cppreference.com/w/cpp/language/implicit_conversion#Qualification_conversions) - the other direction is not allowed though.

Reply:
As well as running genbindings on every version of Qt and diffing/merging the results, the other step would be to run it for clang-linux, clang-mingw, and clang-osxcross to identify all the cross-platform functions (#59).